packages/sdk/src/assets/nft.ts

```diff
@@ -1,21 +1,37 @@
-import {numberToHex} from 'viem'
+import {Address} from 'abitype'
+import {Hex, numberToHex, pad, toHex} from 'viem'
 
+import {GAS_CONTROLLER} from '../constants/gas-controllers'
 import {NotMintedNftError} from '../errors/assets/not-minted-nft.error'
+import {CreateNft, DEFAULT_TOKEN_URI, HolographNFTMetadata, validate} from './nft.validation'
+import {Config} from '../services'
+import {queryTokenIdFromReceipt} from '../utils/decoders'
 import {IsNotMinted} from '../utils/decorators'
-import {HolographNFTMetadata, validate} from './nft.validation'
+import {HolographConfig, TokenUriType} from '../utils/types'
+import {CxipERC721, HolographDropERC721} from '../contracts'
 
 export class NFT {
   isMinted: boolean
   private metadata: HolographNFTMetadata
-  private owner?: string
-  private chainId?: string | number
+  private chainId: number
+  private collectionAddress?: Address
   private file?: string
+  private owner?: string
   private tokenId?: string // Decimal tokenId string
-  private collectionAddress?: string
-  private mintTx?: string
+  private txHash?: string
 
-  constructor(metadata: HolographNFTMetadata) {
-    this.metadata = metadata
+  private cxipERC721: CxipERC721
+  private holographDropERC721: HolographDropERC721
+
+  constructor(configObject: HolographConfig, {chainId, collectionAddress, metadata}: CreateNft) {
+    this.chainId = validate.chainId.parse(chainId)
+    this.collectionAddress = validate.collectionAddress.parse(collectionAddress) as Address
+    this.metadata = validate.metadata.parse(metadata)
+    const config = Config.getInstance(configObject)
+    const cxipERC721 = new CxipERC721(config, this.collectionAddress)
```

Comment:
I do not like that the nft has both kinds of collections defined within it. I can see that a dev can confuse if this nft suppose to be minted to a legacy collection or an MOE collection.

Reply:
but we have two different methods `mint` and `moeMint`, do you want to create a separate class for the moe nft?

Reply:
Done